tests/h/auth/util_test.py

```diff
@@ -1,56 +1,45 @@
-import pytest
-
-from h.auth import util
+from h.auth.util import client_authority, default_authority
+from h.security import Identity
 
 
 class TestClientAuthority:
-    @pytest.mark.parametrize(
-        "principals",
-        [
-            ["foo", "bar", "baz"],
-            ["authority", "foo"],
-            [],
-            ["client_authority:"],
-            [" client_authority:biz.biz", "foo"],
-            ["client_authority :biz.biz", "foo"],
-        ],
-    )
-    def test_it_returns_None_if_no_client_authority_principal_match(
-        self, principals, pyramid_request, pyramid_config
-    ):
-        pyramid_config.testing_securitypolicy("LYZADOODLE", groupids=principals)
-
-        assert util.client_authority(pyramid_request) is None
-
-    @pytest.mark.parametrize(
-        "principals,authority",
-        [
-            (
-                ["foo", "bar", "baz", "client_authority:felicitous.com"],
-                "felicitous.com",
-            ),
-            (["client_authority:somebody.likes.me", "foo"], "somebody.likes.me"),
-        ],
-    )
-    def test_it_returns_authority_if_authority_principal_matchpyramid_requesi(
-        self, principals, authority, pyramid_request, pyramid_config
-    ):
-        pyramid_config.testing_securitypolicy("LYZADOODLE", groupids=principals)
-
-        assert util.client_authority(pyramid_request) == authority
+    def test_it_with_no_identity(self, pyramid_request, pyramid_config):
+        self.set_identity(pyramid_request, pyramid_config, identity=None)
+
+        result = client_authority(pyramid_request)
+
+        assert result is None
+
+    def test_it_with_auth_client(self, pyramid_request, pyramid_config, factories):
+        identity = Identity(auth_client=factories.AuthClient())
+        self.set_identity(pyramid_request, pyramid_config, identity=identity)
+
+        result = client_authority(pyramid_request)
+
+        assert result == identity.auth_client.authority
+
+    def set_identity(self, pyramid_request, pyramid_config, identity):
+        # This should be simplified after the 2.0 upgrade as the other code
+        # path will not be required
```

Comment:
```suggestion
```

Let's try to avoid TODO comments. A likely result is that the Pyramid 1.x bit below does get deleted in future but we forget to delete this comment a few lines above it, resulting in a baffling comment. The `# Pyramid 1.x` comment below makes it clear that the `except` is no longer needed once we move to Pyramid 2.0. It could be made even more explicit by changing it to something like `# Only needed for Pyramid 1.x compatibility.`

You could make a project board for the Pyramid 2.0 project and add an issue with a link to this function, if you want to make sure we don't forget to simplify it.